### ISSUE TITLE
amstrad/amstrad: MAME fails the majority of the SHAKER tests

### ISSUE THREAD

**Issue by malespiaut** [mamedev/mame#12289]
### MAME version

0.264 (unknown)

### System information

OS: Arch Linux, 64-bit, kernel 6.8.5-zen1-1-zen
Locale: en_US.UTF-8
RAM: 16GB
CPU: 11th Gen Intel i7-1165G7 (8) @ 4.700GHz
GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics]

### Emulated system/software

Amstrad CPC6128 (cpc6128)

### Incorrect behaviour

The SHAKER tests is an all-in-one floppy to test CPC emulator accuracy, developped by the demo group Logon System. It can be downloaded at https://shaker.logonsystem.eu/.

Most tests show graphical glitches, compared to actual “screen shot” of the tests running on actual hardware. All screen shots are listed at https://shaker.logonsystem.eu/tests

### Expected behaviour

MAME should display the same graphical result as the screen shots of the tests running on actual hardware.

### Steps to reproduce

1. Launch mamke for any CPC machine (`cpc464`, `cpc6128`, `cpc464+`, `cp6128+`, etc.)
2. Load the floppy `shaker25.dsk` on slot 1.
3. Type `cat` to list the content of the floppy. Or directly type `run"shake25X` (replace `X` with either `a`, `b`, `c`, or `d`).
4. Choose any test, and compare the result with what is shown at https://shaker.logonsystem.eu/tests (Note: For choices that have several tests, press SPACE to move to the next one. Some tests (i.e. VSYNC) can take time.)

**Comment by holub** [comment]
Amstead is known for lacking accuracy in MAME.
Can you please use https://mametesters.org/ for such reports? At least all be stored in same place.

**Comment by angelosa** [comment]
> Can you please use https://mametesters.org/ for such reports? At least all be stored in same place.

I disagree: sounds like a spike https://github.com/mamedev/mame/issues/10504 and https://github.com/mamedev/mame/issues/9106 , better suited with GH issues. I'll look at this at some point in future, putting labels as reminder.

**Comment by flowstate247** [comment]
I recommend that you read this technical article about the CRTC chips on CPCwiki:
https://www.cpcwiki.eu/index.php/CRTC

If MAME implemented everything described there, CPC emulation would improve a lot.
